### ISSUE TITLE
Missing Fields in proto!

### ISSUE THREAD

**Issue by exterkamp** [GoogleChrome/lighthouse#6362]
#### What is the current behavior?
Lighthouse_result.proto has some missing fields! Doh!
 - [ ] config_settings.audit_mode
 - [ ] Timing

**Comment by patrickhulce** [comment]
Maybe step one before/while adding these, is adding a round trip assertion test? :)

**Comment by exterkamp** [comment]
The reason these were missed was because the sample json cleans these two fields out for testing!  So I totally missed them!

I support round-tripping the smokehouse JSONs or pulling some real world ones and round tripping them.  The diffing gets complicated though.  That has been the problem. @paulirish

**Comment by patrickhulce** [comment]
Oh I didn't realize we were just going to reuse the existing diff infra for it :) 

Since that diffing was made to go between runs where timing/date/etc might end up changing and this one is about detecting *any* data loss on a roundtrip would it might make sense to use a stricter diff for the protobuf?

**Comment by exterkamp** [comment]
Yes, we could do a much stricter diff for protobuf.  I'm doing that kind of strict diffing for LR, but it should be done for LH too.  The current sample roundtrip was always a first step.

**Comment by patrickhulce** [comment]
Oh cool sorry I didn't realize we're already doing it for LR!
